Makefile

```diff
@@ -0,0 +1,38 @@
+ROOT_DIR := $(patsubst %/,%,$(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
+BUILD_DIR := $(ROOT_DIR)/.build
+OUT_DIR := $(ROOT_DIR)/src/proto
+
+KAVA_TAG ?= upgrade-v44
+KAVA_PROTO_DIR = $(BUILD_DIR)/kava/proto
+KAVA_THIRD_PARTY_PROTO_DIR = $(BUILD_DIR)/kava/third_party/proto
+
+NPM_BIN_DIR := $(ROOT_DIR)/node_modules/.bin
+TS_PROTO_PLUGIN_PATH := $(NPM_BIN_DIR)/protoc-gen-ts_proto
+
+.PHONY: all
+all: proto-deps proto-gen
+
+.PHONY: proto-deps
+proto-deps: clean
+	mkdir -p $(BUILD_DIR) && \
+		cd $(BUILD_DIR) && \
+		git clone https://github.com/kava-labs/kava.git --no-checkout --depth 1 --filter=blob:none --sparse && \
+		cd $(BUILD_DIR)/kava && \
+		git sparse-checkout init &&\
+		git sparse-checkout add proto third_party && \
+		git fetch origin $(KAVA_TAG) --depth 1 && \
+		git checkout FETCH_HEAD
+
+.PHONY: proto-gen
+proto-gen:
+	protoc \
```

Comment:
Assumes `protoc` is installed. `brew install protobuf` on osx, and in most linux package managers